framework-plugins/lisk-framework-http-api-plugin/src/controllers/node.ts

```diff
@@ -26,3 +27,56 @@ export const getNodeInfo = (channel: BaseChannel) => async (
 		next(err);
 	}
 };
+
+export const getTransactions = (channel: BaseChannel, codec: PluginCodec) => async (
+	req: Request,
+	res: Response,
+	next: NextFunction,
+): Promise<void> => {
+	const { limit, offset } = req.query;
+	let limitNumber = 10;
+	let offsetNumber = 0;
+
+	if (limit) {
+		if (!isNumberString(limit)) {
+			res.status(400).send({
+				errors: [{ message: 'The limit query parameter should be a number.' }],
+			});
+			return;
+		}
+		limitNumber = Number(limit);
+	}
+
+	if (offset) {
+		if (!isNumberString(offset)) {
+			res.status(400).send({
+				errors: [{ message: 'The offset query parameter should be a number.' }],
+			});
+			return;
+		}
+		offsetNumber = Number(offset);
+	}
+
+	let transactionsInPool;
+
+	try {
+		transactionsInPool = await channel.invoke<string>('app:getTransactionsFromPool');
+	} catch (err) {
+		next(err);
+		return;
+	}
+
+	const decodedTransactions = [];
+	for (const transaction of transactionsInPool) {
+		decodedTransactions.push(codec.decodeTransaction(transaction));
+	}
+
+	// 200 - Response
+	res.status(200).json({
+		data: decodedTransactions.slice(
+			offsetNumber || 0,
+			Math.min(limitNumber + offsetNumber, decodedTransactions.length),
+		),
```

Comment:
maybe slice the `transactionsInPool` collection, that way we could avoid extensive decoding and less iterations too?